backend/service/chaos/experimentation/experimentstore/experiment_store.go

```diff
@@ -58,61 +50,75 @@ func New(cfg *any.Any, _ *zap.Logger, _ tally.Scope) (service.Service, error) {
+	// This API call will eventually be broken into 2 separate calls:
+	// 1) creating the config
+	// 2) starting a new experiment with the config
+
+	// All experiments are created in a single transaction
+	tx, err := fs.db.Begin()
+	if err != nil {
+		return err
+	}
 
 	for _, experiment := range experiments {
-		marshaler := jsonpb.Marshaler{}
-		buf := &bytes.Buffer{}
-		err := marshaler.Marshal(buf, experiment.GetConfig())
+		// Step 1) create the config
+		configID := id.NewID()
+
+		configJson, err := marshalConfig(experiment.GetConfig())
 		if err != nil {
 			return err
 		}
-		s := buf.String()
-		builder = builder.Values(
-			id.NewID(),
-			s,
-		)
-	}
 
-	sql, args, err := builder.ToSql()
-	if err != nil {
-		return err
+		configSql := `INSERT INTO experiment_config (id, details) VALUES ($1, $2)`
+		_, err = fs.db.ExecContext(ctx, configSql, configID, configJson)
+		if err != nil {
+			return err
+		}
+
+		// Step 2) start a new experiment with the config
+		runSql := `
+			INSERT INTO experiment_run (
+                id, 
+		        experiment_config_id,
+		        execution_time,
+		        creation_time)
+            VALUES ($1, $2, tstzrange(NOW(), NULL), NOW())`
+
+		runId := id.NewID()
+		_, err = fs.db.ExecContext(ctx, runSql, runId, configID)
+		if err != nil {
+			return err
+		}
+
+		// TODO(bgallagher) temporarily returning the experiment run ID. Eventually, the CreateExperiments function
+		// will be split into CreateExperimentConfig and CreateExperimentRun in which case they will each return
+		// their respective IDs
+		experiment.Id = uint64(runId)
 	}
 
-	_, err = fs.db.ExecContext(ctx, sql, args...)
+	err = tx.Commit()
 	return err
 }
 
 // DeleteExperiments deletes the specified experiments from the store.
 func (fs *experimentStore) DeleteExperiments(ctx context.Context, ids []uint64) error {
-	psql := sq.StatementBuilder.PlaceholderFormat(sq.Dollar)
-	builder := psql.Delete("").From(experimentsTableName)
-
-	if len(ids) > 0 {
-		builder = builder.Where(sq.Eq{"id": ids})
-	}
-
-	sql, args, err := builder.ToSql()
-	if err != nil {
-		return err
+	if len(ids) != 1 {
+		// TODO: This API will be renamed to StopExperiment and will take a single ID as a parameter
+		return errors.New("A single ID must be provided")
 	}
-
-	_, err = fs.db.ExecContext(ctx, sql, args...)
+	sql := `DELETE FROM experiment_run WHERE id == $1`
+	_, err := fs.db.ExecContext(ctx, sql, ids[0])
 	return err
 }
 
 // GetExperiments gets all experiments
 func (fs *experimentStore) GetExperiments(ctx context.Context) ([]*experimentation.Experiment, error) {
-	psql := sq.StatementBuilder.PlaceholderFormat(sq.Dollar)
-	builder := psql.Select(experimentColumns...).From(experimentsTableName)
+	sql := `
+        SELECT experiment_run.id, details FROM experiment_config, experiment_run
+        WHERE experiment_config.id = experiment_run.experiment_config_id
+        	AND now() <@ experiment_run.execution_time`
```

Comment:
is this condition needed in this test?

Reply:
It's not needed now but will be in the next change. I suppose I could remove it.